### PR TITLE
Detect & fix backwards markdown links (of the form (foo)[bar])

### PR DESCRIPTION
**Pull request by timstclair**

### CODE REVIEW

File: cmd/mungedocs/links.go [hunk 1]
```diff
@@ -29,6 +29,8 @@ import (
 var (
 	// Finds markdown links of the form [foo](bar "alt-text").
 	linkRE = regexp.MustCompile(`\[([^]]*)\]\(([^)]*)\)`)
+	// Finds markdown link typos of the form (foo)[bar]
+	badLinkRE = regexp.MustCompile(`\(([^]()]*)\)\[([^]()]*)\]`)
```
Comment (eparis) [inline cmd/mungedocs/links.go]: I feel like this belongs in util.go and should be more like `mlLinkRE`. And we should annotate the mungeLine with `badLink` or something. Then just check for `badLink` here.  I'm ok doing it here, but as I recall regexps with submatches were measurably slower and should be avoided when they weren't needed. (`mlLinkRE` hits every line, `linkRE` should only hit lines that already passed `mlLinkRE`, so I think that should be the regexp we copy here)

Comment (timstclair) [inline cmd/mungedocs/links.go]: `FindString` and `MatchString` have nearly identical performance, so I removed the capture groups and switched to `FindString`. I think it's cleaner to keep this logic in links.go, since this data seems irrelevant to other mungers.

**Comment by timstclair** [comment]
@k8s-bot test this issue: https://github.com/kubernetes/kubernetes/issues/22187

**Comment by timstclair** [comment]
@k8s-bot test this issue: https://github.com/kubernetes/kubernetes/issues/22187
